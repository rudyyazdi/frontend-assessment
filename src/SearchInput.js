import React, { useState } from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import Loader from 'react-loader-spinner'

import './search-input.css'

const SearchInput = ({ className }) => {
  const [isLoading, setIsLoading] = useState(false)

  return (<div className={classNames([className, 'search-input'])}>
    <input type="text" placeholder="Search.." />
    {isLoading &&
      <div className="loader-spinner">
        <Loader
          type="Oval"
          color="grey"
          height="30"
          width="30"
        />
      </div>
    }
  </div>)
}

SearchInput.propTypes = {
  className: PropTypes.string
}

export default SearchInput
