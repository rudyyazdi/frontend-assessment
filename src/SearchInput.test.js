import React from 'react';
import SearchInput from './SearchInput';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<SearchInput isLoading={true} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
