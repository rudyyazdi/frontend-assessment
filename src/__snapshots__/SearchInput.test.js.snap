// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="search-input"
>
  <input
    onChange={[Function]}
    placeholder="Search.."
    type="text"
    value=""
  />
</div>
`;
